js_src/_dart_helpers.js

```diff
@@ -131,10 +131,21 @@ function _createReactDartComponentClass2(dartInteropStatics, componentStatics, j
       return typeof derivedState !== 'undefined' ? derivedState : null;
     }
     render() {
+      // Must call checkPropTypes manually because React moved away from using the `prop-types` package.
+      // See: https://github.com/facebook/react/pull/18127
+      // React now uses its own internal cache of errors for PropTypes which broke `PropTypes.resetWarningCache()`.
+      // Solution was to use `PropTypes.checkPropTypes` directly which makes `PropTypes.resetWarningCache()` work.
+      // Solution from: https://github.com/facebook/react/issues/18251#issuecomment-609024557
+      // TODO: figure out how to get the `displayName` here...
+      React.PropTypes.checkPropTypes(jsConfig.propTypes, this.props, 'prop', this._getDisplayName());
       var result = dartInteropStatics.handleRender(this.dartComponent, this.props, this.state, this.context);
       if (typeof result === 'undefined') result = null;
       return result;
     }
+    // Hacky workaround to get the displayName for `checkPropTypes` call.
+    _getDisplayName() {
+      return ReactDartComponent2.displayName;
+    }
```

Comment:
It should be noted that I opted for this route to prevent breakages in the `PropTypes` api, specifically the `resetWarningCache()` react has opted to build in their own "prop-types" rather than using the external package. Currently react does not provide a way to reset the warning cache of itself, which is why i opted to use the external package. It is also unclear what the future hold for prop-types in general so the external package is a safer bet for long term support as the React team feels that Typescript on its own provides most of the prop type checks statically, unfortunately dart types do not support things such as Literal Types, Tuples, Unions, Optional/Required fields (i mean they kinda do but not as nicely), etc...